rosbag2_storage_mcap/CMakeLists.txt

```diff
@@ -19,77 +19,26 @@ endif()
 set(ROS_DISTRO $ENV{ROS_DISTRO})
 
 find_package(ament_cmake REQUIRED)
+find_package(ament_index_cpp REQUIRED)
+find_package(mcap_vendor REQUIRED)
 find_package(pluginlib REQUIRED)
 find_package(rcutils REQUIRED)
 find_package(ros_environment REQUIRED)
 find_package(rosbag2_storage REQUIRED)
-find_package(ament_index_cpp REQUIRED)
-
-# Install conan
-find_package(Python3 COMPONENTS Interpreter REQUIRED)
-execute_process(COMMAND ${Python3_EXECUTABLE} -m pip install conan)
-set(ENV{PATH} $ENV{HOME}/.local/bin:$ENV{PATH})
-
-if(NOT EXISTS "${CMAKE_BINARY_DIR}/conan.cmake")
-  message(STATUS "Downloading conan.cmake from https://github.com/conan-io/cmake-conan")
-  file(DOWNLOAD "https://raw.githubusercontent.com/conan-io/cmake-conan/v0.16.1/conan.cmake"
-                "${CMAKE_BINARY_DIR}/conan.cmake"
-                EXPECTED_HASH SHA256=396e16d0f5eabdc6a14afddbcfff62a54a7ee75c6da23f32f7a31bc85db23484
-                TLS_VERIFY ON)
-endif()
-
-include(${CMAKE_BINARY_DIR}/conan.cmake)
-
-include(FetchContent)
-fetchcontent_declare(mcap
-  GIT_REPOSITORY https://github.com/foxglove/mcap.git
-  GIT_TAG 0851e316cf973c26a64773c56540efbbaf06725a
-)
-fetchcontent_makeavailable(mcap)
-
-set(MCAP_ROOT ${CMAKE_CURRENT_BINARY_DIR}/_deps/mcap-src/cpp)
-set(MCAP_INCLUDE_DIRS ${MCAP_ROOT}/mcap/include)
-
-conan_cmake_autodetect(settings)
-
-conan_cmake_install(
-  PATH_OR_REFERENCE ${MCAP_ROOT}/mcap
-  INSTALL_FOLDER ${MCAP_ROOT}/mcap/build/Release
-  BUILD missing
-  SETTINGS compiler=gcc
-  SETTINGS compiler.version=11
-  SETTINGS compiler.libcxx=libstdc++
-  SETTINGS ${settings}
-)
-
-include(${MCAP_ROOT}/mcap/build/Release/conanbuildinfo.cmake)
-conan_basic_setup()
 
 add_library(
   ${PROJECT_NAME} SHARED
   src/mcap_storage.cpp
   src/message_definition_cache.cpp
 )
 target_compile_features(${PROJECT_NAME} PUBLIC c_std_99 cxx_std_17)
-target_include_directories(${PROJECT_NAME}
-  SYSTEM PUBLIC
-    $<BUILD_INTERFACE:${MCAP_INCLUDE_DIRS}>
-    $<INSTALL_INTERFACE:include>
-)
 
 target_link_libraries(${PROJECT_NAME}
-  ${CONAN_LIBS}
+  mcap_vendor::mcap
```

Comment:
@emersonknapp I'm not sure if we want the namespace to be `mcap_vendor::` or `mcap::`, but I tried combinations randomly until this target_link_libraries worked. For some reason, the `install(... NAMESPACE "mcap_vendor::" ...)` didn't seem to affect what I need to write here in order to make it build successfully. Any tips? I'm sure something is wrong with this cmake setup somewhere...